src/game.cpp

```diff
@@ -10199,47 +10199,32 @@ bool game::handle_liquid(item &liquid, bool from_ground, bool infinite, item *so
 
     } else if (liquid.is_ammo() && (cont->is_tool() || cont->is_gun())) {
         // for filling up chainsaws, jackhammers and flamethrowers
```

Comment:
This idiom is repeated several times with near-identical code. A future PR could consider refactoring this further but I'm reluctant to make such extraneous changes in what is already a fairly major change.

Incidental to this PR tools and guns with magazines can now be refilled correctly with liquids.